app/uk/gov/hmrc/cgtpropertydisposals/actors/TaxEnrolmentRetryManager.scala

```diff
@@ -0,0 +1,133 @@
+  createConnectorActor: ActorRefFactory => ActorRef,
+  taxEnrolmentRetryRepository: TaxEnrolmentRetryRepository,
+  taxEnrolmentService: TaxEnrolmentService,
+  scheduler: Scheduler
+) extends Actor
+    with Logging {
+  import context.dispatcher
+  implicit val hc: HeaderCarrier = HeaderCarrier()
+
+  val config: Config = ConfigFactory.load()
+
+  val minBackoff: FiniteDuration = config.as[FiniteDuration]("tax-enrolment-retry.min-backoff")
+  val maxBackoff: FiniteDuration = config.as[FiniteDuration]("tax-enrolment-retry.max-backoff")
+  val numberOfRetriesUntilInitialWaitDoubles: Int =
+    config.as[Int]("tax-enrolment-retry.number-of-retries-until-initial-wait-doubles")
+  val maxAllowableElapsedTime: FiniteDuration = config.as[FiniteDuration]("tax-enrolment-retry.max-elapsed-time")
+
+  override def preStart(): Unit = {
+    super.preStart()
+
+    val mongoActorRef     = createMongoActor(context)
+    val connectorActorRef = createConnectorActor(context)
+
+    context.watch(mongoActorRef)
+    context.watch(connectorActorRef)
+
+    mongoActorRef ! GetAllNonFailedTaxEnrolmentRetryRequests
+    context.become(initialised(mongoActorRef, connectorActorRef))
+  }
+
+  def initialised(mongo: ActorRef, connector: ActorRef): Receive = {
+    case Terminated(deadActor) => {
+      if (deadActor == mongo) {
+        logger.info("Creating new mongo actor")
+        val newMongoActor = createMongoActor(context)
+        context.watch(newMongoActor)
+        context.become(initialised(newMongoActor, connector))
+      } else if (deadActor == connector) {
+        logger.info("Creating new connector actor")
+        val newConnectorActor = createConnectorActor(context)
+        context.watch(newConnectorActor)
+        context.become(initialised(mongo, newConnectorActor))
+      } else
+        logger.warn(s"Unexpected terminated message - dead actor ref: ${deadActor.toString}")
+    }
+
+    case retry: RetryTaxEnrolmentRequest =>
+      if (retry.taxEnrolmentRequest.currentElapsedTimeInSeconds >= maxAllowableElapsedTime.toSeconds) {
+        logger.info("Elapsed time has exceed maximum allowed time allowed to retry - marking request as failed")
+        mongo ! UpdateTaxEnrolmentRetryStatusToFail(retry.taxEnrolmentRequest)
+      } else {
+        val interval = nextScheduledTime(retry.taxEnrolmentRequest.numOfRetries)
+        val retryRequest = RetryCallToTaxEnrolmentService(
+          retry.taxEnrolmentRequest.copy(
+            numOfRetries                = retry.taxEnrolmentRequest.numOfRetries + 1,
+            timeToNextRetryInSeconds    = interval.toSeconds,
+            currentElapsedTimeInSeconds = retry.taxEnrolmentRequest.currentElapsedTimeInSeconds + retry.taxEnrolmentRequest.timeToNextRetryInSeconds
+          )
+        )
+        logger.info(s"Retrying request with following details: [${retryRequest.taxEnrolmentRequest}]")
+        scheduler.scheduleOnce(interval, connector, retryRequest)
```

Comment:
if the connector actor dies before this message reaches it we're going to end up with dead letters. The simplest way to remedy this is to probably get this actor to schedule a message to itself that when received will immediately send a retry message to the connector actor.